packages/python/plotly/plotly/express/_core.py

```diff
@@ -424,14 +460,19 @@ def make_trace_kwargs(args, trace_spec, trace_data, mapping_labels, sizeref):
                     if len(customdata_cols) > 0:
                         # here we store a data frame in customdata, and it's serialized
                         # as a list of row lists, which is what we want
-                        trace_patch["customdata"] = trace_data[customdata_cols]
+
+                        # dict.fromkeys(customdata_cols) allows to deduplicate column
+                        # names, yet maintaining the original order.
+                        trace_patch["customdata"] = trace_data.select(
+                            [nw.col(c) for c in dict.fromkeys(customdata_cols)]
+                        )
```

Comment:
For pandas, it was possible to have duplicates in the list. This is banned in polars and arrow by default and in pandas under narwhals.